python/ray/serve/config.py

```diff
@@ -170,8 +176,9 @@ def location_backfill_no_server(cls, v, values):
     @validator("root_url", always=True)
     def fill_default_root_url(cls, v, values):
         if v == "":
-            if SERVE_ROOT_URL_ENV_KEY in os.environ:
-                return os.environ[SERVE_ROOT_URL_ENV_KEY]
+            serve_root_url_env = ray.get(get_serve_root_url_env.remote())
+            if serve_root_url_env:
+                return serve_root_url_env
```

Comment:
I'd prefer to just do this in the controller instead, we can just do it out-of-band from the `HTTPOptions` validation.